### PR TITLE
remove tracking annotation from validation and webhooks

### PR DESCRIPTION
**Pull request by kannon92**
#### What type of PR is this?
/kind cleanup

/kind deprecation
#### What this PR does / why we need it:
In 1.27, we removed the legacy tracking job in the job controller.  We no longer check the annotations so we want to now remove validation in 1.28.  
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
remove tracking annotation from validation and defaulting
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
-[KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/2307-job-tracking-without-lingering-pods#deprecation
```

### CODE REVIEW

**Comment by kannon92** [comment]
https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/2307-job-tracking-without-lingering-pods#deprecation states that we are removing the feature toggle.  I also think we should remove the code around validating these annotation or have test cases for it.

**Comment by kannon92** [comment]
@alculquicondor we discussed this offline.

**Comment by alculquicondor** [comment]
ref kubernetes/enhancements#2307

**Review by alculquicondor — APPROVED** [review]
For a change like this, we need to think about it across 2 versions:

1. If a job is created in k8s v1.28, would it pass update validations in v1.27?

In 1.28, it would be possible to create a job with the annotation, because we completely ignore it from validation. As a result, this annotation can have any value.
The validation in 1.27 allows the annotation on update if it already has it. The value of the annotation is irrelevant.

2. If a job is created in k8s v1.28, would it behave the same in v1.27?

In v1.27, jobs are created with the annotation (for backwards compatibility with v1.26), but the job controller actually ignores it.
So a job created in v1.28 without the annotation behaves the same in v1.27. We don't care about n-2 skew.

In conclusion, this PR is safe.

/lgtm
/approve

**Comment by alculquicondor** [comment]
@kannon92 could you update the KEP to reflect that all the deprecation steps where completed? https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/2307-job-tracking-without-lingering-pods
